### ISSUE TITLE
[rocm-llvm] Many tests fail

### ISSUE THREAD

**Issue by kozross** [rocm-arch/rocm-arch#461]
Full list is below. This means the package fails to build.

```
Failed Tests (114):
  LeakSanitizer-Standalone-i386 :: TestCases/Linux/cleanup_in_tsd_destructor.c
  LeakSanitizer-Standalone-i386 :: TestCases/Linux/fork.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/Linux/fork_and_leak.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/Linux/fork_threaded.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/Linux/guard-page.c
  LeakSanitizer-Standalone-i386 :: TestCases/Linux/log-path_test.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/Linux/use_tls_pthread_specific_dynamic.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/Linux/use_tls_pthread_specific_static.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/Linux/use_tls_static.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/default_options.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/do_leak_check_override.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/ignore_object_errors.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/leak_check_at_exit.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/leak_check_before_thread_started.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/link_turned_off.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/new_array_with_dtor_0.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/pointer_to_self.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/print_suppressions.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/recoverable_leak_check.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/register_root_region.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/suppressions_default.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/suppressions_file.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/swapcontext.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/use_after_return.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/use_globals_initialized.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/use_globals_uninitialized.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/use_registers.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/use_stacks.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/use_stacks_threaded.cpp
  LeakSanitizer-Standalone-i386 :: TestCases/use_unaligned.cpp
  Profile-i386 :: Linux/coverage_ctors.cpp
  Profile-i386 :: Linux/coverage_dtor.cpp
  Profile-i386 :: Linux/coverage_test.cpp
  Profile-i386 :: Linux/instrprof-comdat.test
  Profile-i386 :: coverage_comments.cpp
  Profile-x86_64 :: Linux/coverage_ctors.cpp
  Profile-x86_64 :: Linux/coverage_dtor.cpp
  Profile-x86_64 :: Linux/coverage_test.cpp
  Profile-x86_64 :: Linux/instrprof-comdat.test
  Profile-x86_64 :: coverage_comments.cpp
  SanitizerCommon-Unit :: ./Sanitizer-i386-Test/SanitizerLinux.ThreadDescriptorSize
  SanitizerCommon-Unit :: ./Sanitizer-x86_64-Test/SanitizerLinux.ThreadDescriptorSize
  SanitizerCommon-lsan-i386-Linux :: Linux/aligned_alloc-alignment.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/aligned_alloc.c
  SanitizerCommon-lsan-i386-Linux :: Linux/allow_user_segv.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/clock_gettime.c
  SanitizerCommon-lsan-i386-Linux :: Linux/closedir.c
  SanitizerCommon-lsan-i386-Linux :: Linux/crypt_r.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/decorate_proc_maps.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/mallopt.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/mprobe.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/new_delete_test.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/pthread_mutex.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/ptrace.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/pvalloc-overflow.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/signal_send.cpp
  SanitizerCommon-lsan-i386-Linux :: Linux/sysmsg.c
  SanitizerCommon-lsan-i386-Linux :: Linux/timerfd.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/crypt.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/fgetc_ungetc_getc.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/fgets.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/fputc_putc_putchar.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/getc_unlocked.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/getrandom.c
  SanitizerCommon-lsan-i386-Linux :: Posix/getusershell.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/lstat.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/mmap_test.c
  SanitizerCommon-lsan-i386-Linux :: Posix/popen.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/posix_memalign-alignment.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/putc_putchar_unlocked.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/readlink.c
  SanitizerCommon-lsan-i386-Linux :: Posix/readlinkat.c
  SanitizerCommon-lsan-i386-Linux :: Posix/regex.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/setvbuf.cpp
  SanitizerCommon-lsan-i386-Linux :: Posix/uname.c
  SanitizerCommon-lsan-i386-Linux :: allocator_returns_null.cpp
  SanitizerCommon-lsan-i386-Linux :: fopen_nullptr.c
  SanitizerCommon-lsan-i386-Linux :: malloc_hook.cpp
  SanitizerCommon-lsan-i386-Linux :: max_allocation_size.cpp
  SanitizerCommon-lsan-i386-Linux :: options-help.cpp
  SanitizerCommon-lsan-i386-Linux :: options-include.cpp
  SanitizerCommon-lsan-i386-Linux :: options-invalid.cpp
  SanitizerCommon-lsan-i386-Linux :: printf-ldbl.c
  SanitizerCommon-lsan-i386-Linux :: pthread_mutexattr_get.cpp
  SanitizerCommon-lsan-i386-Linux :: sanitizer_coverage_inline8bit_counter.cpp
  SanitizerCommon-lsan-i386-Linux :: sanitizer_coverage_inline_bool_flag.cpp
  SanitizerCommon-lsan-i386-Linux :: sanitizer_coverage_trace_pc_guard-init.cpp
  SanitizerCommon-lsan-i386-Linux :: sanitizer_coverage_trace_pc_guard.cpp
  SanitizerCommon-lsan-i386-Linux :: scanf-ldbl.c
  SanitizerCommon-lsan-i386-Linux :: strcasestr.c
  SanitizerCommon-lsan-i386-Linux :: strcspn.c
  SanitizerCommon-lsan-i386-Linux :: strpbrk.c
  SanitizerCommon-lsan-i386-Linux :: strstr.c
  SanitizerCommon-lsan-i386-Linux :: wcrtomb.c
  SanitizerCommon-lsan-i386-Linux :: wctomb.c
  SanitizerCommon-lsan-x86_64-Linux :: Linux/clock_gettime.c
  SanitizerCommon-lsan-x86_64-Linux :: Linux/closedir.c
  SanitizerCommon-lsan-x86_64-Linux :: Linux/getcpuclockid.c
  SanitizerCommon-lsan-x86_64-Linux :: Linux/mmap64_test.c
  SanitizerCommon-lsan-x86_64-Linux :: Linux/sysmsg.c
  SanitizerCommon-lsan-x86_64-Linux :: Posix/getrandom.c
  SanitizerCommon-lsan-x86_64-Linux :: Posix/mmap_test.c
  SanitizerCommon-lsan-x86_64-Linux :: Posix/readlink.c
  SanitizerCommon-lsan-x86_64-Linux :: Posix/readlinkat.c
  SanitizerCommon-lsan-x86_64-Linux :: printf-ldbl.c
  SanitizerCommon-lsan-x86_64-Linux :: scanf-ldbl.c
  SanitizerCommon-lsan-x86_64-Linux :: strcasestr.c
  SanitizerCommon-lsan-x86_64-Linux :: strcspn.c
  SanitizerCommon-lsan-x86_64-Linux :: strnlen.c
  SanitizerCommon-lsan-x86_64-Linux :: strpbrk.c
  SanitizerCommon-lsan-x86_64-Linux :: strspn.c
  SanitizerCommon-lsan-x86_64-Linux :: strstr.c
  SanitizerCommon-lsan-x86_64-Linux :: wcrtomb.c
  SanitizerCommon-lsan-x86_64-Linux :: wctomb.c


Testing Time: 651.99s
  Unsupported      : 17704
  Passed           : 56041
  Expectedly Failed:   152
  Failed           :   114
```

**Comment by Tzigamm** [comment]
Same here, I canceled the tests it after something like 15 min, they were using 100% CPU and some logging was happening through systemd-coredump from time to time

**Comment by tpkessler** [comment]
I think that failed tests are not related to our `PKGBUILD` script. This issue should be reported upstream. As a workaround, run `makepkg --nocheck`.

**Comment by Tzigamm** [comment]
Is it safe though? Since the rest of the stack depends on it does it cause any error/problem?

**Comment by acxz** [comment]
hmm not sure if it would be safe or not, but I am leaning towards commenting out the unit tests in the PKGBUILD for now. Do you guys think that would be fine?

**Comment by tpkessler** [comment]
I agree with that. For the unit tests, file an issue upstream. It's not a problem of the packaging but of the software itself.

**Comment by Tzigamm** [comment]
Urg, it seems we can't open issues on [their llvm repo](https://github.com/RadeonOpenCompute/llvm-project), no one has raised an issue on [their main repo](https://github.com/RadeonOpenCompute/ROCm) as of now though, even though the new release is 5 days old, is it just that ROCm have so few user no one else ran into it?

**Comment by tpkessler** [comment]
Yes, their LLVM repo is a fork and thus does not have an issue tracker. You can open issue on the main ROCm repo, or more specifically, on the AOMP repo. We can also ask the Gentoo guys if they experience similar difficulties.

**Comment by acxz** [comment]
patched with 70f75fc

**Comment by acxz** [comment]
I'm going ahead and closing this issue as I just tried to add tests back in with the following change:
```
check() {
  cd "$srcdir/llvm-$pkgver.src/build"
  LD_LIBRARY_PATH=$PWD/lib ninja check
}
```

and received the following error:

```
FAILED: projects/compiler-rt/lib/tsan/CMakeFiles/GotsanRuntimeCheck /home/acxz/vcs/git/github/rocm-arch/rocm-arch/rocm-llvm/src/projects/compiler-rt/lib/tsan/CMakeFiles/GotsanRuntimeCheck
cd /home/acxz/vcs/git/github/rocm-arch/rocm-arch/rocm-llvm/src/llvm-project-rocm-5.0.0/compiler-rt/lib/tsan/go && env "CC=/usr/bin/clang " EXTRA_CFLAGS= IN_TMPDIR=1 SILENT=1 /home/acxz/vcs/git/github/rocm-arch/rocm-arch/rocm-llvm/src/llvm-project-rocm-5.0.0/compiler-rt/lib/tsan/go/buildgo.sh
/usr/bin/ld: /tmp/gotsan.CmJRD9GIjB/race_linux_amd64.syso: in function `__clang_call_terminate':
gotsan.cpp:(.text.__clang_call_terminate[__clang_call_terminate]+0x6): undefined reference to `__cxa_begin_catch'
/usr/bin/ld: gotsan.cpp:(.text.__clang_call_terminate[__clang_call_terminate]+0xb): undefined reference to `std::terminate()'
/usr/bin/ld: /tmp/gotsan.CmJRD9GIjB/race_linux_amd64.syso:(.data.DW.ref.__gxx_personality_v0[DW.ref.__gxx_personality_v0]+0x0): undefined reference to `__gxx_personality_v0'
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
[182/889] Building CXX object tools/clang/unittests/Analysis/CMakeFiles/ClangAnalysisTests.dir/CloneDetectionTest.cpp.o
ninja: build stopped: subcommand failed.
```

I don't really think there is a point in tracking unit testing for now.
